### ISSUE TITLE
org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests.testCachedBytesReadsAndWrites Fails Reproducibly

### ISSUE THREAD

**Issue by original-brownbear** [elastic/elasticsearch#55707]
Fails every time for me on master:

```
Suite: Test class org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests
  1> [2020-04-24T16:05:06,871][INFO ][o.e.i.s.SearchableSnapshotDirectoryStatsTests] [testCachedBytesReadsAndWrites] before test
  1> [2020-04-24T16:05:07,029][INFO ][o.e.i.s.SearchableSnapshotDirectoryStatsTests] [testCachedBytesReadsAndWrites] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:searchable-snapshots:test' --tests "org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests.testCachedBytesReadsAndWrites" -Dtests.seed=ADBEADD211855FEE -Dtests.security.manager=true -Dtests.locale=ar-SY -Dtests.timezone=Antarctica/DumontDUrville -Dcompiler.java=14 -Druntime.java=11
  2> java.lang.AssertionError: 
    Expected: <600000000L>
         but: was <1200000000L>
        at __randomizedtesting.SeedInfo.seed([ADBEADD211855FEE:680F1CA0138B987E]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests.lambda$testCachedBytesReadsAndWrites$2(SearchableSnapshotDirectoryStatsTests.java:117)
        at org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests.executeTestCase(SearchableSnapshotDirectoryStatsTests.java:602)
        at org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests.executeTestCaseWithCache(SearchableSnapshotDirectoryStatsTests.java:542)
        at org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests.testCachedBytesReadsAndWrites(SearchableSnapshotDirectoryStatsTests.java:97)
  2> NOTE: leaving temporary files on disk at: /home/brownbear/src/elasticsearch/x-pack/plugin/searchable-snapshots/build/testrun/test/temp/org.elasticsearch.index.store.SearchableSnapshotDirectoryStatsTests_ADBEADD211855FEE-001
  2> NOTE: test params are: codec=Asserting(Lucene84): {}, docValues:{}, maxPointsInLeafNode=1158, maxMBSortInHeap=5.586960239552283, sim=Asserting(org.apache.lucene.search.similarities.AssertingSimilarity@7d98b1be), locale=ar-SY, timezone=Antarctica/DumontDUrville
  2> NOTE: Linux 5.4.0-26-generic amd64/Oracle Corporation 11.0.2 (64-bit)/cpus=16,threads=1,free=461190184,total=536870912
  2> NOTE: All tests run in this JVM: [SearchableSnapshotDirectoryStatsTests]

```

**Comment by tbrooks8** [comment]
This failed against on my (unrelated) PR merge (https://gradle-enterprise.elastic.co/s/t4lyapratj55q).

**Comment by tlrx** [comment]
@tbrooks8 the test failure you had concerned the test `SearchableSnapshotDirectoryStatsTests.testReadBytesContiguously` which is different from the one in the issue. Anyway, the test has been fixed in #58912 which has been merged after your PR (so your PR did not contain the fix at the time it hit the failure).
